Repositories/TodoRepository/WhenContextHasData.cs

```diff
@@ -0,0 +1,137 @@
+
+            Assert.NotNull(todo);
+        }
+
+        [Fact]
+        public void GivenValidEntity_Update_UpdateEntity()
+        {
+            var repo = new Xml.TodoRepository(MockXmlContext.Object);
+            var guid = new Guid("20975aeb-d490-4aa6-95ba-5b7c50b074a4");
+            var todo = repo.GetById(guid);
+
+            todo.Title = "Concert";
+            todo.Description = "Go to Metallica concert";
+            repo.Update(todo);
+            var newTodo = repo.GetById(guid);
+
+            Assert.NotNull(newTodo);
+            Assert.Equal(todo.Title, newTodo.Title);
+            Assert.Equal(todo.Description, newTodo.Description);
+        }
+
+        [Theory]
+        [InlineData("Picnic","Go to a picnic with friends",TodoStatus.Open,"2020-05-15T14:29:15.1823029Z","2020-05-19T21:00:00.0000000Z")]
+        [InlineData("Football", "", TodoStatus.InProgress, "2020-05-15T14:29:15.1823029Z", "2020-05-19T21:00:00.0000000Z")] // without Description
+        public void GivenValidTodo_Insert_ReturnsNewTodo(string title,string description, TodoStatus status, string createdOn, string dueDate)
+        {
+            var repo = new Xml.TodoRepository(MockXmlContext.Object);
+
+            var entity = new TodoModel()
+            {
+                Title = title,
+                Description = description,
+                Status = status,
+                CreatedOn = DateTime.Parse(createdOn),
+                DueDate = DateTime.Parse(dueDate)
+            };
+
+            var todo = repo.Insert(entity);
+            
+            Assert.NotNull(todo);
+        }
+
+        [Theory]
+        [InlineData("Picnic", "Go to a picnic with friends", TodoStatus.Open, "2020-05-15T14:29:15.1823029Z", "2020-05-19T21:00:00.0000000Z")]
+        [InlineData("Football", "", TodoStatus.InProgress, "2020-05-15T14:29:15.1823029Z", "2020-05-19T21:00:00.0000000Z")] // without Description
+        public void GivenValidTodo_Insert_AddsTodo(string title, string description, TodoStatus status, string createdOn, string dueDate)
+        {
+            var repo = new Xml.TodoRepository(MockXmlContext.Object);
+
+            var entity = new TodoModel()
+            {
+                Title = title,
+                Description = description,
+                Status = status,
+                CreatedOn = DateTime.Parse(createdOn),
+                DueDate = DateTime.Parse(dueDate)
+            };
+
+            var todo = repo.Insert(entity);
+            
+            var guid = todo.Id;
+            var addedTodo = repo.GetById(guid);
+
+            Assert.NotNull(addedTodo);
+        }
+
+        [Fact]
+        public void GivenValidElement_Delete_RemoveElement()
+        {
+            var repo = new Xml.TodoRepository(MockXmlContext.Object);
+
+            var guid = new Guid("00000000-0000-0000-0000-000000000000");
+            repo.Delete(guid);
+
+            var todo = repo.GetById(guid);
+
+            Assert.Null(todo);
+        }
+    }
```

Comment:
Add unit tests for `Get`:
- `GivenFilter_Get_ReturnsMatchingTodos`
- `GivenOrderBy_Get_ReturnsSortedTodos`
- `GivenFilterAndOrderBy_Get_ReturnsMatchingSortedTodos`

Example usages of `Get`:
```cs
repo.Get(todo => todo.Status == TodoStatus.Open); // filter - get todos with Status = Open
repo.Get(todo => todo.Title); // order by Title
repo.Get(todo => todo.Status == TodoStatus.Open, todo => todo.Title); // filter by Status = Open AND sort by Title
```

For each test add different type of filters and order by conditions (`Theory`).
Because lambda expressions are not allowed to be used directly in InlineData, you can use MemberData with [Tuple](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-tuples). Other ways of achieving this can be found here: https://stackoverflow.com/questions/45299470/lambda-expression-as-inline-data-in-xunit

Examples:
```cs
public static TheoryData<(Func<TodoModel, bool>, string[])> FilterTests = new TheoryData<(Func<TodoModel, bool> filter, string[] eptected)>
{
    ( todo => todo.Status == TodoStatus.Open, new string[] { "00000000-0000-0000-0000-000000000000", "20975aeb-d490-4aa6-95ba-5b7c50b074a4"} ),
    ( todo => todo.Title == "Football", new string[] { "20975aeb-d490-4aa6-95ba-5b7c50b074a4"} ),
};

[Theory]
[MemberData(nameof(FilterTests))]
public void GivenFilter_Get_ReturnsMatchingTodos((Func<TodoModel, bool> filter, string[] expectedIds) data)
{
    // Arrange
    var repo = new Xml.TodoRepository(MockXmlContext.Object);

    // Act
    var result = repo.Get(data.filter);

    // Assert
    Assert.Equal(data.expectedIds.Length, result.Count());

    foreach (var expectedId in data.expectedIds)
    {
        Assert.Contains(result, todo => todo.Id.ToString() == expectedId);
    }
}

public static TheoryData<(Func<TodoModel, object>, string[])> OrderByTests = new TheoryData<(Func<TodoModel, object> orderByKey, string[] eptected)>
{
    ( todo => todo.Title, new string[] { "20975aeb-d490-4aa6-95ba-5b7c50b074a4", "00000000-0000-0000-0000-000000000000" } ),
};

[Theory]
[MemberData(nameof(OrderByTests))]
public void GivenOrderBy_Get_ReturnsSortedTodos((Func<TodoModel, object> orderByKey, string[] expectedIds) data)
{
    // Arrange
    var repo = new Xml.TodoRepository(MockXmlContext.Object);

    // Act
    var result = repo.Get(data.orderByKey);

    // Assert
    Assert.Equal(data.expectedIds.Length, result.Count());

    for (int i = 0; i < data.expectedIds.Length; i++)
    {
        var expectedId = data.expectedIds[i];
        var todo = result.ElementAt(i);
        Assert.Equal(expectedId, todo.Id.ToString());
    }
}
```